### ISSUE TITLE
election-provider-multi-phase: Add extrinsic to challenge signed submissions

### ISSUE THREAD

**Issue by emostov** [paritytech/substrate#11051]
The election provider multi phase pallet allows "staking miners" to submit elections solutions during the signed phase. We only store a limited number of solutions. Due to the current storage layout, for any given solution score, only one solution can exist. Currently we check the best solution, and if its ok we don't check the remaining solutions. All of the previous create a situation where a spammer can submit bad solutions without little to no negative repercussions. 

To address this we should add a challenge extrinsic. This extrinsic would take a solution id and check if the solution matches its claimed score. If it does not the challenger would receive a reward and the solution submitter would lose their deposit. Otherwise the challenger would be slashed some large amount to compensate for the cost of checking the solution. 

The  challengers reward should be less than the deposit of the miner in order to not give a spammer the incentive to submit bad solutions and challenge themselves.

**Comment by Doordashcon** [comment]
Hi @emostov, i've been going through the code and would like to take on this issue :)

**Comment by emostov** [comment]
Sounds good - let me know if you have any questions

**Comment by Doordashcon** [comment]
"solution id" means submission index and this index can be used to retrieve a signed submission?

**Comment by emostov** [comment]
Yes, it should be the submission index that is used as the key to this map: 
`pub type SignedSubmissionsMap<T: Config> = StorageMap<_, Twox64Concat, u32, SignedSubmissionOf<T>, OptionQuery>;`

**Comment by kianenigma** [comment]
> Due to the current storage layout, for any given solution score, only one solution can exist, which is problematic if miners converge on the same score since an attacker can block virtually all honest miners simply pre-crafting that score and submitting it with a bogus solution. This means the cost of ensuring no signed submissions works is just the deposit of a single solution as opposed to the cost of a queues worth of deposits.


FYI, this is no longer the case, as of https://github.com/paritytech/substrate/pull/12237
